bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairIntelliChlorHandler.java

```diff
@@ -34,91 +37,156 @@
  *
  * @author Jeff James - Initial contribution
  */
+@NonNullByDefault
 public class PentairIntelliChlorHandler extends PentairBaseThingHandler {
     private final Logger logger = LoggerFactory.getLogger(PentairIntelliChlorHandler.class);
 
-    protected PentairPacketIntellichlor pic3cur = new PentairPacketIntellichlor();
-    protected PentairPacketIntellichlor pic4cur = new PentairPacketIntellichlor();
+    protected int version;
+    protected String name = "";
+
+    /** for a saltoutput packet, represents the salt output percent */
+    public int saltOutput;
+    /** for a salinity packet, is value of salinity. Must be multiplied by 50 to get the actual salinity value. */
+    public int salinity;
+
+    public boolean ok;
+    public boolean lowFlow;
+    public boolean lowSalt;
+    public boolean veryLowSalt;
+    public boolean highCurrent;
+    public boolean cleanCell;
+    public boolean lowVoltage;
+    public boolean lowWaterTemp;
+    public boolean commError;
+
+    public static @Nullable PentairIntelliChlorHandler onlineChlorinator;
```

Comment:
I don't think you should use a static field to check for uniqueness in this case. I think a better approach would be to check whether or not any other chlorinators already exist on the bridge instead.

Reply:
> For future code changes to the PR, please make them additional commits. Please don't append to the current commit and then force push. It is getting very difficult to track what changes you are making in response to our reviews. Thank you.

No problem.  Last time during submission the request was to squash call changes into a single commit.  Is that something that will be done before the merge?

Reply:
The PR will automatically be squashed as part of merging. So you don't need to do anything.